### ISSUE TITLE
Bug: Cannot log URLs

### ISSUE THREAD

**Issue by abirtley** [fullstack-build/tslog#247]
**Describe the bug**
If you attempt to log a URL, you get a variety of errors, depending on whether the URL is a top level logged item, or a field inside a logged object. Either way, it fails.

**To Reproduce**

```ts
import { Logger } from 'tslog';
process.version; //? 18.16.0
const logger = new Logger({});
logger.info(new URL('https://www.abc.def')); //? Cannot convert undefined or null to object
logger.info({ u: new URL('https://www.abc.def') }); //? TypeError: Cannot convert undefined or null to object at getPrototypeOf (<anonymous>) at URL.[nodejs.util.inspect.custom...
```

**Expected behavior**
The URLs are logged without error.

**Screenshots**

<img width="737" alt="image" src="https://github.com/fullstack-build/tslog/assets/19584280/4528a557-6279-4a58-915f-f785670e2eea">
<img width="980" alt="image" src="https://github.com/fullstack-build/tslog/assets/19584280/2684ce40-fd11-456b-8aec-de6caa7513f6">

**Additional context**

Seems to be a reversion of https://github.com/fullstack-build/tslog/issues/93, but perhaps a different cause.

I have managed to hack around it with the following additions in BaseLogger.js (node_modules/tslog/dist/esm/BaseLogger.js). With this amended code, URL objects can be logged without issue.

```js
    _recursiveCloneAndMaskValuesOfKeys(source, keys, seen = []) {
        if (seen.includes(source)) {
            return { ...source };
        }
        if (typeof source === "object" && source != null) {
            seen.push(source);
        }

// NEW FUNCTION isURL - JUST A PROOF OF CONCEPT REALLY
        function isUrl(candidate) {
            if (!candidate || typeof candidate !== 'object') {
                return false;
            }
            if (candidate.href !== undefined && candidate.hash !== undefined && candidate.pathname !== undefined) {
                return true;
            }
            return false;
        }


        return isBuffer(source)
            ? source
            : source instanceof Map
                ? new Map(source)
                : source instanceof Set
                    ? new Set(source)
                    : Array.isArray(source)
                        ? source.map((item) => this._recursiveCloneAndMaskValuesOfKeys(item, keys, seen))
                        : source instanceof Date
                            ? new Date(source.getTime())
                            : isError(source)
                                ? Object.getOwnPropertyNames(source).reduce((o, prop) => {
                                    o[prop] = keys.includes(this.settings?.maskValuesOfKeysCaseInsensitive !== true ? prop : prop.toLowerCase())
                                        ? this.settings.maskPlaceholder
                                        : this._recursiveCloneAndMaskValuesOfKeys(source[prop], keys, seen);
                                    return o;
                                }, this._cloneError(source))
                                // NEW CODE: IF THIS IS A URL, JUST RETURN IT AS A STRING. DO NOT RECURSE.
                                : isUrl(source)
                                ? source.toString()
                                : source != null && typeof source === "object"
                                    ? Object.getOwnPropertyNames(source).reduce((o, prop) => {
                                        o[prop] = keys.includes(this.settings?.maskValuesOfKeysCaseInsensitive !== true ? prop : prop.toLowerCase())
                                            ? this.settings.maskPlaceholder
                                            : this._recursiveCloneAndMaskValuesOfKeys(source[prop], keys, seen);
                                        return o;
                                    }, Object.create(Object.getPrototypeOf(source)))
                                    : ((source) => {
                                        this.settings?.maskValuesRegEx?.forEach((regEx) => {
                                            source = source?.toString()?.replace(regEx, this.settings.maskPlaceholder);
                                        });
                                        return source;
                                    })(source);
```

**Node.js Version**
e.g. 18.16.0

**OS incl. Version**
macOS 13.4.1 (c)

**Comment by alexanderguy** [comment]
I came here to report the same issue with `tslog v4.8.2`, on both OS X and Linux with two different versions of node.  Running the above test case, here are the stack traces I'm seeing:

```
node:internal/url:668
        ObjectGetPrototypeOf(this[context]) !== URLContext.prototype) {
        ^

TypeError: Cannot convert undefined or null to object
    at getPrototypeOf (<anonymous>)
    at [nodejs.util.inspect.custom] (node:internal/url:668:9)
    at formatValue (node:internal/util/inspect:806:19)
    at inspect (node:internal/util/inspect:365:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2273:40)
    at formatWithOptions (node:internal/util/inspect:2135:10)
    at transportFormatted (file://redacted/node_modules/tslog/dist/esm/runtime/nodejs/index.js:89:33)
    at Logger.log (file://redacted/node_modules/tslog/dist/esm/BaseLogger.js:104:19)
    at Logger.info (file://redacted/node_modules/tslog/dist/esm/index.js:20:22)
    at file://redacted/repro.mjs:4:8

Node.js v18.13.0


node:internal/url:815
    return this.#context.href;
                ^

TypeError: Cannot read private member #context from an object whose class did not declare it
    at get href [as href] (node:internal/url:815:17)
    at [nodejs.util.inspect.custom] (node:internal/url:756:21)
    at formatValue (node:internal/util/inspect:804:19)
    at inspect (node:internal/util/inspect:363:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2297:40)
    at formatWithOptions (node:internal/util/inspect:2159:10)
    at transportFormatted (file://redacted/node_modules/.pnpm/tslog@4.8.2/node_modules/tslog/dist/esm/runtime/nodejs/index.js:89:33)
    at Logger.log (file://redacted/node_modules/.pnpm/tslog@4.8.2/node_modules/tslog/dist/esm/BaseLogger.js:104:19)
    at Logger.info (file://redacted/node_modules/.pnpm/tslog@4.8.2/node_modules/tslog/dist/esm/index.js:20:22)
    at file://redacted/repro.mjs:4:8

Node.js v20.2.0
```

**Comment by terehov** [comment]
It's solved in `V4.9.0`.
Thanks for pointing it out.

**Comment by PAStheLoD** [comment]
Hello,

I was not sure whether to open a new issue or not, but in the end I decided it's better to ask here first. So, to the point... the Response returned by NodeJS's fetch() also has private members that tslog tries to access resulting in the same error.


Node v24.0.2
tslog 4.9.3

```javascript
const tslog = await import('tslog')
const log = new tslog.Logger()
const r = await fetch('https://1.1.1.1')

log.debug(r)
```


```javascript
Uncaught:
TypeError: Cannot read private member #state from an object whose class did not declare it
    at get status (node:internal/deps/undici/undici:9660:21)
    at [nodejs.util.inspect.custom] (node:internal/deps/undici/undici:9703:24)
    at formatValue (node:internal/util/inspect:877:19)
    at inspect (node:internal/util/inspect:404:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2397:40)
    at formatWithOptions (node:internal/util/inspect:2259:10)
    at Object.transportFormatted (file:///xxx/node_modules/.pnpm/tslog@4.9.3/node_modules/tslog/dist/esm/runtime/nodejs/index.js:107:33)
    at Logger.log (file:///xxx/node_modules/.pnpm/tslog@4.9.3/node_modules/tslog/dist/esm/BaseLogger.js:101:32)
    at Logger.debug (file:///xxx/node_modules/.pnpm/tslog@4.9.3/node_modules/tslog/dist/esm/index.js:23:22)
```



The really strange part is that `utils.inspect(r, { colors: true, compact: false, depth: Infinity })` works outside tslog, but in [runtime/nodejs/index.ts](https://github.com/fullstack-build/tslog/blob/master/src/runtime/nodejs/index.ts#L151) basically that's what gets called.

It seems some kind of unlucky manipulation happens to the Response object.

Digging deeper it seems the `Object.create(Object.getPrototypeOf(source))` in   `_recursiveCloneAndMaskValuesOfKeys` causes the problem, as doing a `console.log(baseObject)` already triggers the error.

If I understand this correctly it's because private fields are not accessible through the prototype chain. (related Stack Overflow [question](https://stackoverflow.com/questions/74757867/cannot-read-private-member-from-an-object-whose-class-did-not-declare-it))

...


okay, so tl;dr for anyone else who ends up here via some mysterious search engine results: set maskValuesOfKeys to empty array, so 

```javascript
import { Logger } from "tslog"

const logger = new Logger({ maskValuesOfKeys: [] })
```
